library/src/permutation/permutation_solution_impl.hpp

```diff
@@ -61,20 +62,23 @@ namespace hiptensor
         {
         }
 
-        bool initArgs(void const*                     alpha,
-                      void const*                     A,
-                      void*                           B,
-                      std::vector<std::size_t> const& a_lengths,
-                      std::vector<std::size_t> const& a_strides,
-                      const int32_t                   modeA[],
-                      std::vector<std::size_t> const& b_lengths,
-                      std::vector<std::size_t> const& b_strides,
-                      const int32_t                   modeB[],
-                      const hipDataType               typeScalar) override
+        bool initArgs(void const*                        alpha,
```

Comment:
I believe this level of the interface we have gone below API level data structures like tensor descriptors. To uncouple the interface from API, maybe we should add the operators as an argument instead?